### ISSUE TITLE
BANANAS: Bayesian Optimization with Neural Architectures for Neural Architecture Search

### ISSUE THREAD

**Issue by sile** [sile/readings#71]
URL
-----

https://arxiv.org/abs/1910.11858

Abstract
----------

> Neural Architecture Search (NAS) has seen an explosion of research in the past few years. A variety of methods have been proposed to perform NAS, including reinforcement learning, Bayesian optimization with a Gaussian process model, evolutionary search, and gradient descent. In this work, we design a NAS algorithm that performs Bayesian optimization using a neural network model. 
We develop a path-based encoding scheme to featurize the neural architectures that are used to train the neural network model. This strategy is particularly effective for encoding architectures in cell-based search spaces. After training on just 200 random neural architectures, we are able to predict the validation accuracy of a new architecture to within one percent of its true accuracy on average, for popular search spaces. This may be of independent interest beyond Bayesian neural architecture search. 
We test our algorithm on the NASBench (Ying et al. 2019) and DARTS (Liu et al. 2018) search spaces, and we show that our algorithm outperforms other NAS methods including evolutionary search, reinforcement learning, AlphaX, ASHA, and DARTS. Our algorithm is over 100x more efficient than random search, and 3.8x more efficient than the next-best algorithm on the NASBench dataset. As there have been problems with fair and reproducible experimental evauations in the field of NAS, we adhere to the recent NAS research checklist (Lindauer and Hutter 2019) to facilitate NAS research. In particular, our implementation has been made publicly available, including all details needed to fully reproduce our results. 

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
  - Neural architecture search
  - Bayesian optimization
  - Predicting neural network accuracy
- (3) Preliminaries
  - Search space
  - Search strategy
  - Evaluation method
  - Bayesian optimization
- (4) Methodology
  - Architecture featurization via path-based encoding
  - Acquisition function and optimization
  - BANANAS: Bayesian optimization with neural architectures for NAS
- (5) Experiments
  - (5-1) NASBench Experiments
  - (5-2) DARTS search space experiments
- (6) Conclusion and Future Work
- (A) Additional Experiments
  - (A-1) Meta Neural Network Experiments
  - (A-2) NASBench Experiments
- (B) Best practices checklist for NAS research

BibTeX
------

```bibtex
@article{white2019bananas,
  title={BANANAS: Bayesian Optimization with Neural Architectures for Neural Architecture Search},
  author={White, Colin and Neiswanger, Willie and Savani, Yash},
  journal={arXiv preprint arXiv:1910.11858},
  year={2019}
}
```

See Also
----------

- GitHub: https://github.com/naszilla/bananas

**Comment by sile** [comment]
> using Bayesian optimization for NAS has so far required specifying a distance function between neural architectures, in order to deﬁne a surrogate model. This is often a cumbersome task involving tuning hyperparameters of the distance function [17, 15]. Furthermore, Bayesian optimization most commonly uses a Gaussian process (GP) model, and it can be quite challenging to achieve highly accurate prediction performance with GPs, given the potentially high dimensional input architectures [6].
> (1. Introduction)

**Comment by sile** [comment]
> while we demonstrated that our path-based featurization performs very well for the most popular cell-based search spaces, our featurization would have trouble scaling to large cell-based search spaces, as its dimensionality is exponential in the number of nodes per cell.
> (6. Conclusion and Future Work)
